pkg/kubectl/rolling_updater.go

```diff
@@ -572,10 +572,23 @@ func AbortRollingUpdate(c *RollingUpdaterConfig) {
 		c.NewRc.Annotations = map[string]string{}
 	}
 	c.NewRc.Annotations[sourceIdAnnotation] = fmt.Sprintf("%s:%s", c.OldRc.Name, c.OldRc.UID)
-	desiredSize, found := c.OldRc.Annotations[desiredReplicasAnnotation]
-	if found {
-		fmt.Printf("Found desired replicas.")
-		c.NewRc.Annotations[desiredReplicasAnnotation] = desiredSize
+
+	// If the old rc being scaled down has its original replica count stored,
+	// use the original value since the replica count change from old to new
+	// could be asymmetric.
+	originalSize, foundOriginal := tmp.Annotations[originalReplicasAnnotation]
+	if foundOriginal {
+		fmt.Printf("Found original replica count %s", originalSize)
+		c.NewRc.Annotations[desiredReplicasAnnotation] = originalSize
+	} else {
+		// If we don't have the old original size, use the desired count from the
+		// new rc. If the update was asymmetric, this won't be quite right, but we
+		// lack the context to make a better decision.
+		desiredSize, foundDesired := c.OldRc.Annotations[desiredReplicasAnnotation]
+		if foundDesired {
+			fmt.Printf("Found desired replica count %s", desiredSize)
```

Comment:
I think since we now require the original annotation to do the update at all, we should fail fast if we can't find it in order to perform a reliable rollback.

Sounds like we're agreed on failing if original isn't available?